### ISSUE TITLE
Is it possible to use purely CPU for CNN?

### ISSUE THREAD

**Issue by AquifersBSIM** [gnina/gnina#234]
### Issue summary
Just a quick question as per the title. 

1. Is it rather possible to use purely CPU for CNN? 
2. If yes, what would be the repercussions (will it be slower? faster? or will the produced results not be as good as to when GPU was used instead)

Thank you in advance!

### Your system configuration
Operating system:
Compiler:
CUDA version (if applicable):
CUDNN version (if applicable):
Python version:

**Comment by dkoes** [comment]
Yes, it will be slower.  Timings are in the gnina paper https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00522-2

**Comment by Jnelen** [comment]
> Yes, it will be slower. Timings are in the gnina paper https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00522-2

I understand not using a GPU will impact the overall docking time by a lot. However, can multiple cpu-cores still help to speed up the CNN scoring? Or can multiple cores only accelerate the initial steps of pose generation? If so, I suppose it would be more efficient to run 4 separate gnina docking calculations on 1 core, instead of 1 gnina calculation using 4 cpu cores?

**Comment by dkoes** [comment]
Presumably you are doing high throughput docking (docking a lot of ligands) in which case running all your gnina jobs on 1 or 2 cores should give you the best throughput.  This lets your job scheduler load balance the jobs and schedule your jobs on nodes with only a few cores available.

The technical reason to request 2 cores is your scheduler (i.e., SLURM) may count hyperthread contexts as cores.

**Comment by Jnelen** [comment]
> Presumably you are doing high throughput docking (docking a lot of ligands) in which case running all your gnina jobs on 1 or 2 cores should give you the best throughput. This lets your job scheduler load balance the jobs and schedule your jobs on nodes with only a few cores available.
> 
> The technical reason to request 2 cores is your scheduler (i.e., SLURM) may count hyperthread contexts as cores.

I am indeed using SLURM. I haven't tried running gnina with 2 cores per job before, but I might play around with that. Thanks for the suggestion! To understand the reasoning slightly better however, what exactly do you mean with hyperthread contexts here?

**Comment by dkoes** [comment]
Modern processors can run  more simultaneous threads than they have cores, see https://www.intel.com/content/www/us/en/gaming/resources/hyper-threading.html

This is only relevant if you have hyperthreading enabled on your cluster nodes.

**Comment by AquifersBSIM** [comment]
Thank you so much @dkoes for the insightful elaboration! Additionally, thank you so much @Jnelen for contributing valuable insights that enhances the discussion! :))
